<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>How is Bioinformatics practiced? · Rajesh Detroja</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="## How is bioinformatics practiced?"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="How is Bioinformatics practiced? · Rajesh Detroja"/><meta property="og:type" content="website"/><meta property="og:url" content="https://rajesh-detroja.github.io/home/"/><meta property="og:description" content="## How is bioinformatics practiced?"/><meta property="og:image" content="https://rajesh-detroja.github.io/home/img/undraw_online.svg"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://rajesh-detroja.github.io/home/img/undraw_tweetstorm.svg"/><meta name="robots" content="noindex"/><link rel="shortcut icon" href="/home/img/favicon.ico"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/xt256.min.css"/><link rel="stylesheet" href="/home/css/code-block-buttons.css"/><link rel="stylesheet" href="/home/css/profile.css"/><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js"></script><script type="text/javascript" src="/home/js/code-block-buttons.js"></script><script src="/home/js/scrollSpy.js"></script><link rel="stylesheet" href="/home/css/prism.css"/><link rel="stylesheet" href="/home/css/main.css"/><script src="/home/js/codetabs.js"></script></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/home/"><img class="logo" src="/home/img/favicon.ico" alt="Rajesh Detroja"/><h2 class="headerTitleWithLogo">Rajesh Detroja</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class=""><a href="/home/index" target="_self">Home</a></li><li class=""><a href="/home/docs/profile/about" target="_self">Profile</a></li><li class=""><a target="_self"></a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><div class="hamburger-menu"><div class="line1"></div><div class="line2"></div><div class="line3"></div></div></div><h2><i>›</i><span>Getting Started</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">Bash<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_1">What is a bash script?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_2">How to start with Bash?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_3">Files and Folders Part 1</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_4">Files and Folders Part 2</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_5">Files and Folders Part 3</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">R<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Getting Started</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/intro">Introduction</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/install">Installation</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/hello-world">Hello World</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/comments">Comments</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/data-types">Data Types</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/variables">Variables</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/operators">Operators</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/decision-making">Decision Making</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/loops">Loops</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/functions">Functions</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/strings">Strings</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/vectors">Vectors</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/lists">Lists</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/matrices">Matrices</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/arrays">Arrays</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/factors">Factors</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/data_frames">Data Frames</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/packages">Packages</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/data_reshaping">Data Reshaping</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Data Interfaces</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/CSV_files">CSV Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/EXCEL_files">EXCEL Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/binary_files">Binary Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/XML_files">XML Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/JSON_files">JSON Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/web_data">Web Data</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/database">Database</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Charts &amp; Graphs</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/pie_charts">Pie Charts</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/bar_charts">Bar Charts</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/boxplots">Boxplots</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/histograms">Histograms</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/line_graphs">Line Graphs</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/scatterplots">Scatterplots</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/venn_diagram">Venn Diagram</a></li></ul></div></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">Python<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Getting Started</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/intro">Introduction</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/install">Installation</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/hello_world">Hello World</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/variables">Variables</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/operators">Operators</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/decision_making">Decision making</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/loops">Loops</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/numbers">Numbers</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/strings">Strings</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/lists">Lists</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/dictionary">Dictionary</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/date_time">Date &amp; Time</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/functions">Functions</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/exceptions">Exceptions</a></li></ul></div></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">BNF<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Getting Started</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/why-bioinformatics">Why Bioinformatics?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/what-is-bioinformatics">What is Bioinformatics?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/biology-for-bioinformaticians">Biology for Bioinformaticians</a></li><li class="navListItem navListItemActive"><a class="navItem" href="/home/docs/BNF/getting_started/how-is-bioinformatics-practiced">How is Bioinformatics practiced?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/how-to-solve-it">How to solve it</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/how-not-to-waste-your-time">How not to waste your time</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Installation</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/computer-setup">How to set up your computer</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/install-conda">How to install conda</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/install-software">How to install software</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/choose-and-install-a-text-editor">Choose and install a text editor</a></li></ul></div></ul></div></div></section></div><script>
            var coll = document.getElementsByClassName('collapsible');
            var checkActiveCategory = true;
            for (var i = 0; i < coll.length; i++) {
              var links = coll[i].nextElementSibling.getElementsByTagName('*');
              if (checkActiveCategory){
                for (var j = 0; j < links.length; j++) {
                  if (links[j].classList.contains('navListItemActive')){
                    coll[i].nextElementSibling.classList.toggle('hide');
                    coll[i].childNodes[1].classList.toggle('rotate');
                    checkActiveCategory = false;
                    break;
                  }
                }
              }

              coll[i].addEventListener('click', function() {
                var arrow = this.childNodes[1];
                arrow.classList.toggle('rotate');
                var content = this.nextElementSibling;
                content.classList.toggle('hide');
              });
            }

            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              var headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                var el = event.target;
                while(el !== headings){
                  if (el.tagName === 'A') {
                    document.body.classList.remove('tocActive');
                    break;
                  } else{
                    el = el.parentNode;
                  }
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer docsContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 id="__docusaurus" class="postHeaderTitle">How is Bioinformatics practiced?</h1></header><article><div><span><h2><a class="anchor" aria-hidden="true" id="how-is-bioinformatics-practiced"></a><a href="#how-is-bioinformatics-practiced" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>How is bioinformatics practiced?</h2>
<p>Bioinformatics requires a broad skill set. The diverse tasks undertaken by bioinformaticians can be roughly divided into three tiers:</p>
<h3><a class="anchor" aria-hidden="true" id="1-data-management"></a><a href="#1-data-management" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>1. Data management.</h3>
<p>Data management requires accessing, combining, converting, manipulating, storing, and annotating data. It requires routine data quality checks, summarizing large amounts of information, and automating existing methods.</p>
<h3><a class="anchor" aria-hidden="true" id="2-primary-data-analysis"></a><a href="#2-primary-data-analysis" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>2. Primary data analysis.</h3>
<p>Analysis of the data requires running alignments, variation callers, and RNA-Seq quantification, as well as finding lists of genes. Strategically, the bioinformatician must anticipate potential pitfalls of planned analyses, find alternatives, and adapt and customize the methods to the data.</p>
<h3><a class="anchor" aria-hidden="true" id="3-data-interpretation"></a><a href="#3-data-interpretation" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>3. Data interpretation.</h3>
<p>Data management and analysis are meaningless without accurate and insightful interpretation. Bioinformaticians discover or support biological hypotheses via the results of their analyses, and so they must be able to interpret their findings in the context of ongoing scientific discourse.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-the-recommended-computer-for-bioinformatics"></a><a href="#what-is-the-recommended-computer-for-bioinformatics" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is the recommended computer for bioinformatics?</h2>
<p>In our experience, the most productive setup for a bioinformatician is using a Mac OS based computer to develop and test the methods and then using a high-performance Linux workstation––or cluster––to execute these pipelines on the data. With the release of Windows 10 Creator Edition (accessible for users that are part of the Windows Insider program), a fully functional Linux Bash shell can be installed and run on Windows. See the <strong>Computer Setup</strong> page for more details.</p>
<h2><a class="anchor" aria-hidden="true" id="how-much-computing-power-do-we-need"></a><a href="#how-much-computing-power-do-we-need" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>How much computing power do we need?</h2>
<p>Bioinformatics methodologies are improving at a rapid pace. A regular workstation is often all you need to analyze data from a typical, high-volume sequencing run (hundreds of millions of measurements). For example, you could handle most RNA-Seq data analysis required for a given day by using only the <code>hisat2</code> aligner on a standard iMac computer (32GB RAM and eight cores used in parallel). More substantial analyses like genome assembly, however, typically require more of memory than what is generally available on a standard desktop computer.</p>
<p>As a rule, low-quality data (contamination, incorrect sample prep, etc.) takes substantially longer to analyze than high-quality data.</p>
<h2><a class="anchor" aria-hidden="true" id="does-learning-bioinformatics-need-massive-computing-power"></a><a href="#does-learning-bioinformatics-need-massive-computing-power" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Does learning bioinformatics need massive computing power?</h2>
<p><strong>No!</strong></p>
<p>Mastering bioinformatics requires nothing more than a standard, UNIX-ready laptop. You can learn bioinformatics using only a $100 Chromebook.</p>
<p>Of course, the computational needs for applying the bioinformatics methods on particular problems will depend on the amount and scale of data being analyzed. A $100 Chromebook has the tools one needs but probably doesn’t have the compute power, storage or memory to run analyses of the massive datasets that may need to be processed.</p>
<p>You might be surprised, however, just how much can be done on just a Chromebook!</p>
<p>Remember this; you don’t need an expensive or powerful system to become one of the best bioinformaticians on the planet! You can learn, practice and gain a profound understanding of this entire field on any simple, Unix capable computing device.</p>
<h2><a class="anchor" aria-hidden="true" id="what-about-the-cloud"></a><a href="#what-about-the-cloud" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What about the cloud?</h2>
<p>Cloud computing is becoming an increasingly common platform for bioinformatics, in particular for efforts that allow bioinformaticians to “bring the tools to the data” - applying your algorithms to massive sequence databases. Cloud services such as Amazon Web Services also enable anyone to host web applications and services on powerful, accessible machines while only paying for the compute time they use.</p>
<p>Running a cloud service involves learning about object stores, virtual private clouds, file systems, and security. Building and maintaining a cloud instance requires you become something like a part-time systems administrator, which can distract you from learning bioinformatics. In later chapters, we will introduce both cloud computing and cloud-based stacks - commonly referred to as software-as-a-service (SaaS), platform-as-a-service (PaaS, and infrastructure-as-a-service (IaaS). These offerings, combined with <em>containerized</em> applications and analyses, have become valued for their potential to provide a scalable platform for reproducible research.</p>
<p>If you are comfortable with these concepts, a cloud-based service can be a companion to this guide, but learning bioinformatics <em>does not require</em> the cloud.</p>
<h2><a class="anchor" aria-hidden="true" id="do-i-need-to-know-unix-to-do-bioinformatics"></a><a href="#do-i-need-to-know-unix-to-do-bioinformatics" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Do I need to know Unix to do bioinformatics?</h2>
<p>Bioinformatics has primarily been developed via freely available tools written on the Unix platform. The vast majority of new advances are published with software written for Unix-based operating systems.</p>
<p>It’s unlikely you’ll be in a position for long-term career advancement as a bioinformatician without a basic understanding of the command line. Starting with 2016, the Windows platform has begun to offer the so-called “Bash for Windows” application that allows the Windows 10 operating system to run almost all of the Unix-specific software.</p>
<p>The good news is that using Unix is not that complicated. While somewhat unforgiving, the Unix philosophy is logical and precise. In our experience, even people with no prior computational background can become confident Unix users in a matter of weeks.</p>
<h2><a class="anchor" aria-hidden="true" id="do-i-need-to-learn-a-programming-language"></a><a href="#do-i-need-to-learn-a-programming-language" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Do I need to learn a programming language?</h2>
<p>Yes –– an introductory level of programming ability (in any programming language) will be necessary.</p>
<p>Fortunately, programming languages function by similar underlying logic –– even when they seem entirely different on the surface. While symbols, syntax, and procedures may differ among programming languages, an introductory level of programming ability in any language is necessary to understand the thought processes required for computational analysis.</p>
<p>We do cover programming concepts in this book. See the section titled <strong>Some programming required</strong> and the corresponding subsections for a primer.</p>
<h2><a class="anchor" aria-hidden="true" id="are-there-alternatives-to-using-unix"></a><a href="#are-there-alternatives-to-using-unix" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Are there alternatives to using Unix?</h2>
<p>Alternatives to Unix are limited but do exist. They fall into two categories:</p>
<h3><a class="anchor" aria-hidden="true" id="software-that-provides-a-web-interface-to-the-command-line-tools"></a><a href="#software-that-provides-a-web-interface-to-the-command-line-tools" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Software that provides a web interface to the command line tools</h3>
<p>These software systems run the same command line tools that one could install into a Unix-based system, but the interfaces to these tools are graphical and provide better information “discoverability.”</p>
<ol>
<li>Galaxy (open source)</li>
<li>GenePattern (open source)</li>
<li>BaseSpace (commercial)</li>
<li>DNA Nexus (commercial)</li>
<li>Seven Bridges Genomics (commercial)</li>
</ol>
<p>Systems that offer custom implementations of bioinformatics methods.</p>
<p>These can be standalone software that runs on your local system –– they are compatible with web-based applications. They often run open-source tools via a GUI.</p>
<ol>
<li>CLC Genomics Workbench</li>
<li>Partek</li>
<li>Golden Helix</li>
<li>SoftGenetics</li>
<li>DNA Star</li>
<li>Geneious</li>
</ol>
<h2><a class="anchor" aria-hidden="true" id="what-is-bioconductor"></a><a href="#what-is-bioconductor" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is Bioconductor?</h2>
<p><a href="https://www.bioconductor.org/">Bioconductor</a> is an open-source software project for the analysis and comprehension of genomic data. Bioconductor is based primarily on the <a href="https://www.r-project.org/">R statistical programming language</a></p>
<p>Users of Bioconductor typically write scripts using R that make calls to functions defined in the Bioconductor library. For example, a Bioconductor-based script that runs an RNA-Seq data analysis could look like this:</p>
<pre><code class="hljs css language-r">biocLite<span class="token punctuation">(</span><span class="token string">"DESeq"</span><span class="token punctuation">)</span>
library<span class="token punctuation">(</span>DESeq<span class="token punctuation">)</span>
count <span class="token operator">=</span> read.table<span class="token punctuation">(</span><span class="token string">"stdin"</span><span class="token punctuation">,</span> header<span class="token operator">=</span><span class="token boolean">TRUE</span><span class="token punctuation">,</span> row.names<span class="token operator">=</span><span class="token number">1</span> <span class="token punctuation">)</span>
cond1 <span class="token operator">=</span> c<span class="token punctuation">(</span><span class="token string">"control"</span><span class="token punctuation">,</span> <span class="token string">"control"</span><span class="token punctuation">,</span> <span class="token string">"control"</span><span class="token punctuation">)</span>
cond2 <span class="token operator">=</span> c<span class="token punctuation">(</span><span class="token string">"treatment"</span><span class="token punctuation">,</span> <span class="token string">"treatment"</span><span class="token punctuation">,</span> <span class="token string">"treatment"</span><span class="token punctuation">)</span>
conds <span class="token operator">=</span> factor<span class="token punctuation">(</span>c<span class="token punctuation">(</span>cond1<span class="token punctuation">,</span> cond2<span class="token punctuation">)</span><span class="token punctuation">)</span>
cdata <span class="token operator">=</span> newCountDataSet<span class="token punctuation">(</span>count<span class="token punctuation">,</span> conds<span class="token punctuation">)</span>
esize <span class="token operator">=</span> estimateSizeFactors<span class="token punctuation">(</span>cdata<span class="token punctuation">)</span>
edisp <span class="token operator">=</span> estimateDispersions<span class="token punctuation">(</span>esize<span class="token punctuation">)</span>
rdata <span class="token operator">=</span> nbinomTest<span class="token punctuation">(</span>edisp<span class="token punctuation">,</span> <span class="token string">"control"</span><span class="token punctuation">,</span> <span class="token string">"treatment"</span><span class="token punctuation">)</span>
write.table<span class="token punctuation">(</span>rdata<span class="token punctuation">,</span> file<span class="token operator">=</span><span class="token string">""</span><span class="token punctuation">,</span> sep<span class="token operator">=</span><span class="token string">"\t"</span><span class="token punctuation">,</span> row.name<span class="token operator">=</span><span class="token boolean">FALSE</span><span class="token punctuation">,</span> quote<span class="token operator">=</span><span class="token boolean">FALSE</span><span class="token punctuation">)</span>
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="what-is-galaxy"></a><a href="#what-is-galaxy" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is Galaxy?</h2>
<p><a href="www.usegalaxy.org">Galaxy</a> is a web-based scientific workflow and data analysis platform that aims to make computational biology accessible to research scientists with no computer programming experience.</p>
<p>Galaxy can generate and present a web-based user interface to existing command-line tools, making them available to those unable to run the tools themselves.</p>
<p><strong>Important:</strong> Galaxy is not a data analysis tool! It is an <strong>Interface</strong> and a <strong>Platform</strong> that provides computational capacity that runs <strong>command line</strong> bioinformatics software on the users’ behalf. Galaxy makes it seem simple to launch tools and manage data – but it is the user’s responsibility to understand the underlying concepts and use these tools correctly.</p>
<p><strong>Pet peeve:</strong> <em>Please don’t say I used Galaxy to analyze data.</em> That is like saying I used a computer to do bioinformatics. As stated above Galaxy is an interface and a platform that uses tools developed by scientists around the world. Recognize the tools authors, cite the publications for the software, then mention and recognize Galaxy.</p>
<p>Galaxy can also serve as graduation moment in a bioinformaticians’ life. We’ve never met an expert bioinformatician that would voluntarily want to use Galaxy for their own needs. You see the “ease” of use of Galaxy is, in the end, nothing more than an <em>illusion.</em> A skilled scientist will always prefer to specify their work via set words such as:</p>
<pre><code class="hljs css language-bash">bwa mem file1.fq file2.fq <span class="token operator">|</span> samtools view -F <span class="token number">68</span> <span class="token operator">|</span> samtools <span class="token function">sort</span> <span class="token operator">></span> result.bam
</code></pre>
<p>instead of describing it as a series of clicks in an interface.</p>
<p>In our opinion, your comfort level of using Galaxy relative to using the command line quantifies your skill level. The moment you feel the drag, the slowdown, the limitations and that “straight-jacket” feel on your creativity that Galaxy imposes on you is the moment where you have graduated from the <em>Elementary School of Galaxy</em> and you’ve entered the <em>Academy of the Command Line.</em></p>
<p>But by all means, if Galaxy works for you keep using it, it is a progression step.</p>
<p>And let’s make one thing clear, the above sentiment is valid for just about all “easy to use” graphical bioinformatics data analysis platforms. There may be a time when you too will recognize that “easy-to-use” is, in reality, a restriction, a curse, it means that you are limited in what you can do and that you will have to operate inside of a glass box with invisible walls.</p>
<p>Rant over :-). Back to our regularly scheduled content.</p>
<p>Galaxy uses a three-panel system where the left panel lists the tools, the middle panel describes the tools, and the right panel displays user data.</p>
<p><u><strong>Galaxy Interface</strong></u></p>
<p><img align="center" src="/home/docs/assets/img/BNF/galaxy_screenshot.png"></img></p>
<h2><a class="anchor" aria-hidden="true" id="what-is-basespace"></a><a href="#what-is-basespace" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is BaseSpace?</h2>
<p><a href="https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub.html">BaseSpace</a> is a cloud-based genomics analysis and storage platform provided by Illumina that directly integrates with all Illumina sequencers.</p>
<p>The system can operate only with sequencing data produced by Illumina and only when the data is in the formats generated by the sequencers.</p>
<h2><a class="anchor" aria-hidden="true" id="are-commercial-bioinformatics-software-packages-expensive"></a><a href="#are-commercial-bioinformatics-software-packages-expensive" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Are commercial bioinformatics software packages expensive?</h2>
<p>From an individual, consumer-level perspective, the cost of commercial bioinformatics software is high –– typically measured in thousands of dollars. The price tag means that few people could or would buy them directly.</p>
<p>For larger organizations, the software costs may pay off quickly in productivity gains as long as the software is chosen carefully to match specific research needs.</p>
<p>Most academicians and open-source developers treat commercial software with some distrust. In our opinion, this distrust may be unwarranted. Many commercial tools cater to specific needs and need to be operated by individuals with diverse levels of training. Also, the goals and resource allocations in academia are typically very different than those of private organizations.</p>
<h2><a class="anchor" aria-hidden="true" id="should-i-freelance-as-a-bioinformatician"></a><a href="#should-i-freelance-as-a-bioinformatician" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Should I freelance as a bioinformatician?</h2>
<p>In our opinion, the answer is <strong>no</strong> unless the terms are generous or your clients appear to recognize the potential complexities that you may run into.</p>
<p>Successful bioinformatics analysis requires more than skill alone: the information content of the data is a crucial second piece. The world’s best bioinformatician won’t be able to extract information that is not present in the data and may have trouble demonstrating that they have done their best.</p>
<p>Our experiences line up with the observations in <strong><a href="https://www.nature.com/news/core-services-reward-bioinformaticians-1.17251">Core services: Reward bioinformaticians, Nature, (2015)</a></strong> that state that analyses always end up being substantially more complicated than anticipated.</p>
<p>It is unexpectedly difficult to estimate the effort that a project requires accurately.</p>
<blockquote>
<p><strong>Amusing story</strong></p>
</blockquote>
<blockquote>
<p>I once got called up by a company offering custom RNA-Seq analysis. It was a quite basic service as far as analyses go. They ran the Tuxedo pipeline and provided the files it generated. Once we established what the service consisted of I asked how much it cost. The answer was surprising:</p>
</blockquote>
<blockquote>
<p>“For educational institutions, we charge <em>two fifty</em> per sample.”</p>
</blockquote>
<blockquote>
<p>I was both stupefied and elated by this answer. Here I am, I thought I knew all about how to estimate the cost of bioinformatics analyses. Yet I am unable to decide what <em>two fifty</em> meant. A typical RNA-Seq at that time might have had ten samples. So:</p>
</blockquote>
<blockquote>
<p><strong>1.</strong> Was it $2.50 per sample, that would sum up to a ridiculously low $25, that adds up to less than the cost of owning the computing power for an analysis of this type.</p>
</blockquote>
<blockquote>
<p><strong>2.</strong> Or was it $250 per sample, that would sum up to a ridiculously high $2500, for executing a straightforward command line tool that at most would take an hour to set up and maybe some hours to run.</p>
</blockquote>
<blockquote>
<p>I meekly inquired: <em>What is two fifty? Is it $2.50?</em></p>
</blockquote>
<blockquote>
<p>*** glacial pause **</p>
</blockquote>
<blockquote>
<p>Answer: <em>No it is $250</em></p>
</blockquote>
<p>Moral of the story, if someone pays $250 per sample, you should take the job. If someone pays $2.50 then better don’t.</p>
<h2><a class="anchor" aria-hidden="true" id="what-do-bioinformaticians-look-like"></a><a href="#what-do-bioinformaticians-look-like" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What do bioinformaticians look like?</h2>
<p>Handsome of course!</p>
<p><img align="center" src="/home/docs/assets/img/BNF/rajesh_detroja.jpg"></img></p>
</span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/home/docs/BNF/getting_started/biology-for-bioinformaticians"><span class="arrow-prev">← </span><span>Biology for Bioinformaticians</span></a><a class="docs-next button" href="/home/docs/BNF/getting_started/how-to-solve-it"><span>How to solve it</span><span class="arrow-next"> →</span></a></div></div></div><nav class="onPageNav"><ul class="toc-headings"><li><a href="#how-is-bioinformatics-practiced">How is bioinformatics practiced?</a><ul class="toc-headings"><li><a href="#1-data-management">1. Data management.</a></li><li><a href="#2-primary-data-analysis">2. Primary data analysis.</a></li><li><a href="#3-data-interpretation">3. Data interpretation.</a></li></ul></li><li><a href="#what-is-the-recommended-computer-for-bioinformatics">What is the recommended computer for bioinformatics?</a></li><li><a href="#how-much-computing-power-do-we-need">How much computing power do we need?</a></li><li><a href="#does-learning-bioinformatics-need-massive-computing-power">Does learning bioinformatics need massive computing power?</a></li><li><a href="#what-about-the-cloud">What about the cloud?</a></li><li><a href="#do-i-need-to-know-unix-to-do-bioinformatics">Do I need to know Unix to do bioinformatics?</a></li><li><a href="#do-i-need-to-learn-a-programming-language">Do I need to learn a programming language?</a></li><li><a href="#are-there-alternatives-to-using-unix">Are there alternatives to using Unix?</a><ul class="toc-headings"><li><a href="#software-that-provides-a-web-interface-to-the-command-line-tools">Software that provides a web interface to the command line tools</a></li></ul></li><li><a href="#what-is-bioconductor">What is Bioconductor?</a></li><li><a href="#what-is-galaxy">What is Galaxy?</a></li><li><a href="#what-is-basespace">What is BaseSpace?</a></li><li><a href="#are-commercial-bioinformatics-software-packages-expensive">Are commercial bioinformatics software packages expensive?</a></li><li><a href="#should-i-freelance-as-a-bioinformatician">Should I freelance as a bioinformatician?</a></li><li><a href="#what-do-bioinformaticians-look-like">What do bioinformaticians look like?</a></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="copyright">Copyright © 2021 Rajesh Detroja</section></footer></div><script>window.fbAsyncInit = function() {FB.init({appId:'351771359141773',xfbml:true,version:'v2.7'});};(function(d, s, id){var js, fjs = d.getElementsByTagName(s)[0];if (d.getElementById(id)) {return;}js = d.createElement(s); js.id = id;js.src = '//connect.facebook.net/en_US/sdk.js';fjs.parentNode.insertBefore(js, fjs);}(document, 'script','facebook-jssdk'));
                </script></body></html>