<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>Biology for Bioinformaticians · Rajesh Detroja</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="## Biology for bioinformaticians"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="Biology for Bioinformaticians · Rajesh Detroja"/><meta property="og:type" content="website"/><meta property="og:url" content="https://rajesh-detroja.github.io/home/"/><meta property="og:description" content="## Biology for bioinformaticians"/><meta property="og:image" content="https://rajesh-detroja.github.io/home/img/undraw_online.svg"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://rajesh-detroja.github.io/home/img/undraw_tweetstorm.svg"/><meta name="robots" content="noindex"/><link rel="shortcut icon" href="/home/img/favicon.ico"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/xt256.min.css"/><link rel="stylesheet" href="/home/css/code-block-buttons.css"/><link rel="stylesheet" href="/home/css/profile.css"/><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js"></script><script type="text/javascript" src="/home/js/code-block-buttons.js"></script><script src="/home/js/scrollSpy.js"></script><link rel="stylesheet" href="/home/css/prism.css"/><link rel="stylesheet" href="/home/css/main.css"/><script src="/home/js/codetabs.js"></script></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/home/"><img class="logo" src="/home/img/favicon.ico" alt="Rajesh Detroja"/><h2 class="headerTitleWithLogo">Rajesh Detroja</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class=""><a href="/home/index" target="_self">Home</a></li><li class=""><a href="/home/docs/profile/about" target="_self">Profile</a></li><li class=""><a target="_self"></a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><div class="hamburger-menu"><div class="line1"></div><div class="line2"></div><div class="line3"></div></div></div><h2><i>›</i><span>Getting Started</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">Bash<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_1">What is a bash script?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_2">How to start with Bash?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_3">Files and Folders Part 1</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_4">Files and Folders Part 2</a></li><li class="navListItem"><a class="navItem" href="/home/docs/bash/bash_5">Files and Folders Part 3</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">R<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Getting Started</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/intro">Introduction</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/install">Installation</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/hello-world">Hello World</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/comments">Comments</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/data-types">Data Types</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/variables">Variables</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/operators">Operators</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/decision-making">Decision Making</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/loops">Loops</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/functions">Functions</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/strings">Strings</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/vectors">Vectors</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/lists">Lists</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/matrices">Matrices</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/arrays">Arrays</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/factors">Factors</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/data_frames">Data Frames</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/packages">Packages</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/getting_started/data_reshaping">Data Reshaping</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Data Interfaces</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/CSV_files">CSV Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/EXCEL_files">EXCEL Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/binary_files">Binary Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/XML_files">XML Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/JSON_files">JSON Files</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/web_data">Web Data</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/data_interfaces/database">Database</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Charts &amp; Graphs</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/pie_charts">Pie Charts</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/bar_charts">Bar Charts</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/boxplots">Boxplots</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/histograms">Histograms</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/line_graphs">Line Graphs</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/scatterplots">Scatterplots</a></li><li class="navListItem"><a class="navItem" href="/home/docs/R/charts_graphs/venn_diagram">Venn Diagram</a></li></ul></div></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">Python<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Getting Started</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/intro">Introduction</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/install">Installation</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/hello_world">Hello World</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/variables">Variables</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/operators">Operators</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/decision_making">Decision making</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/loops">Loops</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/numbers">Numbers</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/strings">Strings</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/lists">Lists</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/dictionary">Dictionary</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/date_time">Date &amp; Time</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/functions">Functions</a></li><li class="navListItem"><a class="navItem" href="/home/docs/Python/getting_started/exceptions">Exceptions</a></li></ul></div></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle collapsible">BNF<span class="arrow"><svg width="24" height="24" viewBox="0 0 24 24"><path fill="#565656" d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"></path><path d="M0 0h24v24H0z" fill="none"></path></svg></span></h3><ul class="hide"><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Getting Started</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/why-bioinformatics">Why Bioinformatics?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/what-is-bioinformatics">What is Bioinformatics?</a></li><li class="navListItem navListItemActive"><a class="navItem" href="/home/docs/BNF/getting_started/biology-for-bioinformaticians">Biology for Bioinformaticians</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/how-is-bioinformatics-practiced">How is Bioinformatics practiced?</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/how-to-solve-it">How to solve it</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/getting_started/how-not-to-waste-your-time">How not to waste your time</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">Installation</h4><ul><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/computer-setup">How to set up your computer</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/install-conda">How to install conda</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/install-software">How to install software</a></li><li class="navListItem"><a class="navItem" href="/home/docs/BNF/installation/choose-and-install-a-text-editor">Choose and install a text editor</a></li></ul></div></ul></div></div></section></div><script>
            var coll = document.getElementsByClassName('collapsible');
            var checkActiveCategory = true;
            for (var i = 0; i < coll.length; i++) {
              var links = coll[i].nextElementSibling.getElementsByTagName('*');
              if (checkActiveCategory){
                for (var j = 0; j < links.length; j++) {
                  if (links[j].classList.contains('navListItemActive')){
                    coll[i].nextElementSibling.classList.toggle('hide');
                    coll[i].childNodes[1].classList.toggle('rotate');
                    checkActiveCategory = false;
                    break;
                  }
                }
              }

              coll[i].addEventListener('click', function() {
                var arrow = this.childNodes[1];
                arrow.classList.toggle('rotate');
                var content = this.nextElementSibling;
                content.classList.toggle('hide');
              });
            }

            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              var headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                var el = event.target;
                while(el !== headings){
                  if (el.tagName === 'A') {
                    document.body.classList.remove('tocActive');
                    break;
                  } else{
                    el = el.parentNode;
                  }
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer docsContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 id="__docusaurus" class="postHeaderTitle">Biology for Bioinformaticians</h1></header><article><div><span><h2><a class="anchor" aria-hidden="true" id="biology-for-bioinformaticians"></a><a href="#biology-for-bioinformaticians" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Biology for bioinformaticians</h2>
<p>Biology is a domain of the Life Sciences, which include, but are not limited to, organic chemistry, ecology, botany, zoology, physiology, and so on. Biology seeks to understand the structures, functions, origins, interactions, and taxonomies of living organisms.</p>
<p>As a science, biology is still relatively immature. It is eclectic, encompassing several scientific domains that are each insufficiently understood and described.</p>
<p>Organisms are immensely complex and always changing. For centuries, we haven’t had precise enough tools to measure, describe, or understand the full extent of their complexity. Digital technologies are changing this. Bioinformatics is at the frontier of these changes, and its potential contributions to Biology and the Life Sciences more broadly are quite exciting. As bioinformatics and other innovative methodologies advance, we expect Life Sciences to mature and develop rich, accurate vocabularies and models to understand and describe living organisms.</p>
<p>For these reasons, current concepts and definitions in the Life Sciences are still just approximations. While they can be sufficiently accurate in some contexts, they may be woefully inadequate in others. In what almost sounds like a “scientific fight club”, the “rules” of bioinformatics are as follows:</p>
<ol>
<li>There are no “universal” rules.</li>
<li>Every seemingly fundamental paradigm has one or more exceptions.</li>
<li>The validity of a bioinformatics method depends on unknown characteristics of the data.</li>
<li>Biology is always more complicated than you think, <em>even after taking this rule into account.</em></li>
</ol>
<p>Below, we have attempted to describe the biological concepts that we deem important to understand the types of information encoded in data. Each of our definitions is short and terse, and we recommend additional self-guided study of each concept of interest.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-dna"></a><a href="#what-is-dna" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is DNA?</h2>
<p><strong><a href="https://en.wikipedia.org/wiki/DNA">DNA</a></strong> stands for Deoxyribo Nucleic Acid.</p>
<p>It is a macromolecule (a molecule constructed of smaller molecules) that carries the genetic instructions required for the development, functioning and reproduction of all known living organisms. In eukaryotic organisms (like animals, plants, and fungi), DNA is present in the nucleus of each cell. In prokaryotic organisms (single-celled organisms like bacteria and mitochondria), DNA is present in the cell’s cytoplasm.</p>
<h3><a class="anchor" aria-hidden="true" id="what-is-dna-composed-of"></a><a href="#what-is-dna-composed-of" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is DNA composed of?</h3>
<p>DNA is made up of two strands of smaller molecules coiled around each other in a double-helix structure. If you uncoiled DNA, you could imagine it looking somewhat like a ladder. Ladders have two important parts: the poles on either side and the rungs that you climb. The “poles” of DNA are made of alternating molecules of deoxyribose (sugar) and phosphate. While they provide the structure, it’s the “rungs” of the DNA that are most important for bioinformatics.</p>
<p>To understand the “rungs” of the DNA, imagine a ladder split in half down the middle so that each half is a pole with a bunch of half-rungs sticking out. In DNA, these “half-rungs” are a molecule called nucleotides. Genetic information is encoded into DNA by the order, or sequence, in which these nucleotides occur. The “ladder” of DNA is held together by the bonds between each “half-rung.”</p>
<h2><a class="anchor" aria-hidden="true" id="what-are-nucleotides"></a><a href="#what-are-nucleotides" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What are nucleotides?</h2>
<p>Nucleotides are the building blocks of nucleic acids (DNA and RNA–we’ll get to that one later on). In DNA, there are four types of nucleotide: Adenine, Cytosine, Guanine, and Thymine. Because the order in which they occur encodes the information biologists try to understand, we refer to them by their first letter, <code>A</code>, <code>C</code>, <code>G</code> and <code>T</code>, respectively.</p>
<pre><code class="hljs css language-text">A   Adenine
C   Cytosine
G   Guanine
T   Thymine
</code></pre>
<p>Back to the ladder analogy. “Nucleotide” is the <em>type</em> of molecule that makes up each half-rung of our ladder. Because they function as the units that encode genetic information, each letter is also called a base.</p>
<p>For an example of how we represent the sequence of DNA bases using these letters, if we took the DNA of the <em>Bacillus anthracis</em> bacteria that causes Anthrax disease, unfolded the double-helix into a ladder, and then split the ladder in two, the top half (the forward strand–we’ll get to that in a bit) would be written like this:</p>
<pre><code class="hljs css language-text">ATATTTTTTCTTGTTTTTTATATCCACAAACTCTTTT
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="what-are-base-pairs"></a><a href="#what-are-base-pairs" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What are base pairs?</h2>
<p>When you put both sides of the ladder back together, each rung is made by the bond between two bases. When they are bonded, we call them a base pair (or sometimes “bp”).</p>
<p>When it comes to base pairs, it’s important to remember that Adenine only ever bonds with Thymine, and Guanine with Cytosine. In the ladder analogy, this means that each rung will be denoted as “A-T,” “T-A,” “G-C,” or “C-G.”</p>
<p>Certain groups of nucleotides that share some common attribute may be designated by so called <em>ambiguity</em> codes, for example, <code>W</code> stands for <code>A</code> or <code>T</code>:</p>
<pre><code class="hljs css language-text">Y   Pyrimidine (C or T)
R   Purine (A or G)
W   Weak (A or T)
S   Strong (G or C)
K   Keto (T or G)
M   Amino (C or A)
D   A, G, T (not C - remember as after C)
V   A, C, G (not T - remember as after T/U - We'll get to "U" soon)
H   A, C, T (not G - remember as after G)
B   C, G, T (not A - remember as after A)
N   Any base
-   Gap
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="what-are-dna-strands"></a><a href="#what-are-dna-strands" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What are DNA strands?</h2>
<p>Remember when we split the DNA “ladder” in half down the middle so that each side was a pole with half-rungs sticking out? These two halves are called strands. To distinguish the two strands, scientists label one the forward strand and the second, the reverse strand. Above, we gave the example of the forward strand in the Bacillus anthracis bacteria. Here’s what it looks like paired with its reverse strand:</p>
<pre><code class="hljs css language-text">forward --> ATATTTTTTCTTGTTTTTTATATCCACAAACTCTTTT
            |||||||||||||||||||||||||||||||||||||
            TATAAAAAAGAACAAAAAATATAGGTGTTTGAGAAAA <-- reverse
</code></pre>
<p>The lines that connect the bases on either side denote a basepair relationship.</p>
<p>“Forward” and “reverse” are just labels. The choice of labeling is arbitrary and does not depend on any inherent property of the DNA. The forward strand is not “special.” Scientists decide which to call “forward” and which to call “reverse” when they first analyze the DNA of an organism. Even though the decision is arbitrary, it’s important to maintain consistency with that decision for the sake of clear communication.</p>
<p>The forward and reverse strands may also be denoted with different terms. For example, in some datasets you may find them labeled as + and -. They might also be called top and bottom strands, or even Watson and Crick strands.</p>
<p>In our opinion, these variances are needlessly confusing. Please avoid referring to strands with any other terms than forward and reverse.</p>
<h2><a class="anchor" aria-hidden="true" id="is-there-a-directionality-of-dna"></a><a href="#is-there-a-directionality-of-dna" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Is there a directionality of DNA?</h2>
<p>Yes, there is a directionality to the DNA determined by the polarity of the molecules. This direction runs in opposite ways for each strand. Typically, we indicate this polarity with arrows:</p>
<pre><code class="hljs css language-text">------>
ATATTTTTTCTTGTTTTTTATATCCACAAACTCTTTT
|||||||||||||||||||||||||||||||||||||
TATAAAAAAGAACAAAAAATATAGGTGTTTGAGAAAA
                              <------
</code></pre>
<p>Most biological mechanisms (but not all) take place on a single strand of the DNA and in the direction of the arrow. Hence, sequences of the DNA above will be “seen” by the biochemical machinery as either:</p>
<pre><code class="hljs css language-text">ATATTTTTTCTTGTTTTTTATATCCACAAACTCTTTT
</code></pre>
<p>or as:</p>
<pre><code class="hljs css language-text">AAAAGAGTTTGTGGATATAAAAAACAAGAAAAAATAT
</code></pre>
<p>This last sequence is called the <em>reverse complement</em> of the first and is formed by reversing the order of the letters then swapping <code>A</code> for <code>T</code> and swapping <code>C</code> for <code>G</code> (and vice-versa).</p>
<p>Hence a DNA sequence AAACT may need to be considered:</p>
<ul>
<li>in reverse, <code>TCAAA</code></li>
<li>as a complement, <code>TTTGA</code></li>
<li>as a reverse-complement, <code>AGTTT</code></li>
</ul>
<h2><a class="anchor" aria-hidden="true" id="what-is-senseantisense"></a><a href="#what-is-senseantisense" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is sense/antisense?</h2>
<p>When a process occurs in the expected direction, its directionality may be called <em>sense</em>; if it is going against the normal direction, its directionality may be called <em>anti-sense</em>.</p>
<p>It is very important not to collate the concepts of <em>forward/reverse</em> with <em>sense/anti-sense</em>, as these concepts are completely unrelated. The <em>sense/anti-sense</em> is relative to a sequence’s direction; the sequence, in turn, may come from a forward or reverse strand.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-dna-sequencing"></a><a href="#what-is-dna-sequencing" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is DNA sequencing?</h2>
<p>DNA sequencing is the “catch-all” terminology that describes all the processes for identifying the composition of a DNA macromolecule.</p>
<p>The results of a DNA sequencing process are data files stored in an unprocessed format, typically either <code>FASTA</code>, <code>FASTQ</code>, or unaligned <code>BAM</code> (called <code>uBAM</code>) files. Most published papers also store their data in repositories that can be downloaded for reanalysis.</p>
<h2><a class="anchor" aria-hidden="true" id="what-gets-sequenced"></a><a href="#what-gets-sequenced" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What gets sequenced?</h2>
<p>It is essential to note that instruments do not directly sequence the DNA in its original form. Sequencing requires a laboratory process that transforms the original DNA into a so-called “sequencing library” - an artificial construct based on the original DNA. The process of creating this sequencing library introduces a wide variety of limitations and artificial properties into the results. Also, the method of building the sequencing library will also limit the information that can be learned about the original DNA molecule.</p>
<p>Most (perhaps all) life scientists use the term “sequencing” over-generously and are often under the assumption that it produces more precise information than what it can deliver.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-a-genome"></a><a href="#what-is-a-genome" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is a genome?</h2>
<p>A genome is all of an organism’s DNA sequence. Each cell typically contains a copy of the entire genome. More accurately, each cell has one or more nearly identical copies of the genome. Replication is the process of duplicating the genome when a cell divides.</p>
<p>While, due to complementarity, the number of <code>A</code> and <code>T</code> nucleotides and the <code>C</code> and <code>G</code> nucleotides is equal, the relative ratios of <code>AT</code> pairs vs <code>CG</code> pairs may be very different. Some genomes may contain more <code>AT</code> pairs while others may contain more <code>CG</code> pairs.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-a-genomes-purpose"></a><a href="#what-is-a-genomes-purpose" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is a genome’s purpose?</h2>
<p>The genome contains the information that makes the functioning of an organism possible. In cellular organisms, for example, the genome has regions that contain the instructions for making proteins. These are typically called “coding regions”.</p>
<p>The genome may also have regions used to produce molecules other than proteins, as well as regions that regulate the rates by which other processes take place.</p>
<p>All genomes are subject to evolutionary principles. Hence, some (or even substantial) parts of a healthy genome may be non-functional and may no longer serve any obvious purpose. Some percent of a genome may consist of copies of various kinds of interspersed, repeat sequences. At some point, biologists labeled these regions “junk DNA”, but the term has since become a lightning rod of controversy.</p>
<p>Identifying non-functional regions has turned out to be more difficult, and quite more controversial than biologists originally anticipated. The so-called <a href="https://en.wikipedia.org/wiki/C-value#C-value_paradox">C Value Paradox</a> captures the observation that the size of a genome does not directly determine the genome’s complexity.</p>
<h2><a class="anchor" aria-hidden="true" id="how-big-is-a-genome"></a><a href="#how-big-is-a-genome" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>How big is a genome?</h2>
<p>Functional genomes range from as short as the 300 base pairs of the prions that cause the mad-cow disease to as long as the 150 billion base pairs of <code>Paris japonica</code>, a rare, perennial, star-like flower from Japan. It is common to refer to genome sizes in terms of kilo-bases (thousands), mega-bases (millions), and giga-bases (billions).</p>
<p>Here are some other genome sizes:</p>
<ul>
<li>Ebola virus genome: 18 thousand basepairs (18Kb)</li>
<li>E.coli bacteria genome: 4 million basepairs (4Mb)</li>
<li>Baker’s yeast fungus genome: 12 million basepairs (12Mb)</li>
<li>Fruit fly genome: 120 million basepairs (120Mb)</li>
<li>Human genome: 3 billion basepairs (3Gb)</li>
<li>Some salamander species: 120 billion basepairs (120 Gb)</li>
</ul>
<h2><a class="anchor" aria-hidden="true" id="what-is-rna"></a><a href="#what-is-rna" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is RNA?</h2>
<p>Whereas DNA is the blueprint, RNA is a smaller interval of this plan translated into a new kind of molecule. This molecule is similar to DNA in that it differs only owing to a few chemical modifications that change its properties relative to DNA. For example, in RNA, the base <code>T</code> (Thymine) is replaced by the nucleotide <code>U</code> (Uracil).</p>
<p>RNA is a polymeric, single-stranded molecule that often performs a function and is believed to exist transiently. Unlike DNA, there are many classes of RNA: <code>mRNA</code>, <code>tRNA</code>, <code>rRNA</code>, and many others. DNA is continuously present in the cell, whereas RNA degrades quickly over time (minutes).</p>
<h2><a class="anchor" aria-hidden="true" id="how-does-a-genome-function"></a><a href="#how-does-a-genome-function" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>How does a genome function?</h2>
<p>The genome has numerous functions, most of which are too complicated (or insufficiently studied) to describe accurately. Instead, scientists employ a “biological narrative” to describe genomic functions and processes.</p>
<p>The “narrative story” is made very simple by design, often accompanied by a simplified illustration of events: step 1 followed by step 2 followed by step 3, etc. While this helps initially, it also runs of risk of explaining concepts in an oversimplified manner. To give you an example of how this looks in practice, here is a “biological narrative” of what is perhaps the most studied genomic phenomenon: <em>primary mRNA transcription</em> in eukaryotic cells.</p>
<blockquote>
<p>“The cell begins by transcribing a”gene&quot; (see later) into an RNA molecule. After transcription, pieces of the RNA are cut out and discarded in a process called “splicing.” Each discarded piece is referred to as an intron. Each piece between consecutive introns is known as an exon, and the RNA molecule with the introns removed is known as messenger RNA, or mRNA.</p>
</blockquote>
<blockquote>
<p>Part of the cell’s method for identifying introns is the presence of GT and AG, the so-called “splice signals” that usually occur as the first and last dinucleotides of an intron. But the mere presence of these dinucleotides is not sufficient for splicing to occur. Perhaps 35% of human genes are alternatively spliced, meaning that under different circumstances, various combinations of exons are selected.&quot;</p>
</blockquote>
<p>Remember, again, that the above explanation is a simplistic description of a far more complex phenomenon. Any single detail of it opens a series of yet unexplainable questions: why is it that only a small subset of <code>GT</code> <code>AG</code> pairs cause splicing and how come the polymerase that initiates a splicing event at a <code>GT</code> knows that there will be a <code>AG</code> later? Or perhaps it does not know and it is all random - but that would have other implications. You see how the rabbit hole only gets deeper and deeper.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-a-protein"></a><a href="#what-is-a-protein" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is a protein?</h2>
<p>A protein is a three-dimensional macromolecule built from a series of so-called “amino acid” molecules that can form a 3D structure. There are 20 kinds of amino acids that can form a protein; these are labeled as letters in the “alphabet” of protein sequences, where each letter is an amino acid. The &quot;alphabetical order of the amino acids alphabet is:</p>
<pre><code class="hljs css language-text">ARNDCEQGHILKMFPSTWYZ
</code></pre>
<p>Where <code>A</code> stands for Alanine, <code>R</code> for Arginine, <code>N</code> for Asparagine <a href="https://en.wikipedia.org/wiki/DNA_codon_table">and so on</a></p>
<p>Proteins can be described by their sequence alone. But in our current state of understanding, the sequence itself is typically insufficient to fully determine the protein’s 3D structure or function.</p>
<p>Whereas DNA and mRNA usually carry <code>information</code>, proteins are the actual, physical building blocks of life. Every living organism is built out of proteins and functions via the interactions of proteins that are being continuously produced.</p>
<p>A short series (less than 40) of amino acids without a well-defined 3D structure is called a polypeptide (peptides).</p>
<h2><a class="anchor" aria-hidden="true" id="how-are-proteins-made"></a><a href="#how-are-proteins-made" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>How are proteins made?</h2>
<p>The process of reading <em>DNA</em> and creating <em>mRNA</em> out of it is called <em>transcription</em>. Then, in eukaryotes, the <em>mRNA</em> is transported out of the nucleus (to the cell’s cytoplasm), where it is converted to a protein sequence in a process called <em>translation</em>. <strong>Multiple proteins (even hundreds) may be translated from a single mRNA molecule.</strong></p>
<ul>
<li>transcription: DNA –&gt; mRNA</li>
<li>translation: mRNA –&gt; Protein</li>
</ul>
<p>To perform the translation, the mRNA is partitioned into units of three consecutive letters, each called a <code>codon</code>. A <code>codon</code> is then translated via a <a href="https://en.wikipedia.org/wiki/DNA_codon_table">translation table</a> into an <code>amino acid</code>. A protein is a sequence of amino acids.</p>
<p>There is a genetic code for translating a codon into an amino acid. For example, the codon <code>TCA</code> (or <code>UGA</code>if we describe it as an RNA sequence) codes for <code>S</code>, the amino acid <em>Serine</em>. There are 64 combinations of codons but only 20 amino acids. Some codons code for the same amino acid, this is called the <a href="https://en.wikipedia.org/wiki/Codon_degeneracy">codon degeneracy</a>. For example:</p>
<pre><code class="hljs css language-text">CGU, CGC, CGA, CGG, AGA, AGG --> Arginine (Arg/R)
</code></pre>
<p>Most (but not all) amino acids are encoded by more than one codon. A few codons, in most cases <code>TAA</code>, <code>TAG</code> and <code>TGA</code> are the so called stop codons; translation stops when one of them is reached.</p>
<p>The translation process begins with the so-called start codon <code>ATG</code> that corresponds to <code>M</code> (the <em>Methionine</em> amino acid).</p>
<p>For example the subunit B of the Cholera Toxin protein complex responsible for the massive, watery diarrhea characteristic of cholera infection has the following sequence:</p>
<pre><code class="hljs css language-text">MIKLKFGVFFTVLLSSAYAHGTPQNITDLCAEYHNTQIYTLNDKIFSYTESLAGKREMAI
ITFKNGAIFQVEVPGSQHIDSQKKAIERMKDTLRIAYLTEAKVEKLCVWNNKTPHAIAAI
SMAN
</code></pre>
<p>It starts with an <code>M</code>, the Methionine, which as we just stated is also the start codon. You may think all protein sequences should start with <code>M</code>.</p>
<p>But wait, remember how every rule has exceptions? That applies here as well. While the overwhelming majority of proteins start with an <code>M</code> it is possible for this <code>M</code> to get cut off during later processes.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-an-orf"></a><a href="#what-is-an-orf" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is an ORF?</h2>
<p>An <code>ORF</code>, or open reading frame, is a sequence of at least, say, 100 consecutive codons without a stop codon.</p>
<p>It is important also to note that while there is a “standard” translation table, some organisms may use slight variations of it. These are called <a href="https://en.wikipedia.org/wiki/List_of_genetic_codes">genetic codes</a>. When translating from genome to protein, the use of the correct genetic code is essential.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-a-gene"></a><a href="#what-is-a-gene" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is a gene?</h2>
<p>In our experience, the word gene is one of the most misused words in the entirety of biology. It is usually interpreted quite differently by different scientists. Some think of genes as the DNA that codes for proteins; others think genes may include upstream elements, and so on. The “official” definition of the <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0000704">term gene</a> in the Sequence Ontology is</p>
<blockquote>
<p>A region (or regions) that includes all of the sequence elements necessary to encode a functional transcript. A gene may include regulatory regions, transcribed regions and other functional sequence regions.</p>
</blockquote>
<h2><a class="anchor" aria-hidden="true" id="do-genomes-have-other-features"></a><a href="#do-genomes-have-other-features" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Do genomes have other features?</h2>
<p>Genomic regions may have a wide variety of functions. Here are a few often-studied types:</p>
<h3><a class="anchor" aria-hidden="true" id="untranslated-regions"></a><a href="#untranslated-regions" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Untranslated regions</h3>
<p>The region of the mRNA before the start codon (or the corresponding genomic region) is called the <code>5' UTR</code> (5 prime UTR), or untranslated region; the portion from the stop codon to the start of the poly-A tail is the <code>3' UTR</code> (three prime UTR).</p>
<h3><a class="anchor" aria-hidden="true" id="promoter-regions"></a><a href="#promoter-regions" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Promoter regions</h3>
<p>The genomic region just before the <code>5' UTR</code> may contain patterns of nucleotides, called the promoter, that are used to position the molecular machinery that performs the transcription to RNA. For about 60% of human genes, the promoter occurs in a <code>CpG</code> island: a region of DNA where C and G are far more frequent than <code>A</code> and <code>T</code>. Other patterns in the DNA tell the cell when (how frequently and in what tissues) to transcribe the gene; that is, they regulate transcription. A pattern that increases the frequency of transcription operations is an <code>enhancer</code>, while one that decreases the rate is a <code>silencer</code>.</p>
<h3><a class="anchor" aria-hidden="true" id="cpg-islands"></a><a href="#cpg-islands" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>CpG islands</h3>
<p>CpG islands are regions of DNA where a C (cytosine) nucleotide is followed by a G guanine nucleotide in the linear sequence of bases along its 5’ -&gt; 3’ direction. CpG is shorthand for 5’—C—phosphate—G—3’, that is, cytosine and guanine separated a phosphate; phosphate links any two nucleosides together in DNA. Promoter regions may contain CpG islands.</p>
<p>Cytosines in CpG dinucleotides can be methylated. In turn, methylated cytosines within a gene may change the gene’s expression, a mechanism that is part of a larger field of science studying gene regulation, called epigenetics.</p>
<p>Below is a section of a CpG island from chromosome 1 of the human genome. It has 30 CpGs and GC% is 77.14.</p>
<pre><code class="hljs css language-text">>gi|568815597:36306860-36307069 Homo sapiens chromosome 1, GRCh38.p7 Primary Assembly
CGGGGCTCCGGAGAGGCGCGGAGGCCGCGCTGTGCGCGCCGCCGAGGTGAGCGCAAGGGCGGGGACGGGC
GCCGGTGGGCGGGTGCACGGAGCCAGTGCGACCCCGGCGTCTCCGGCTCTTAGTGACGGGCGCGGCTCTG
GGCGGGACCTCGGGGCCGCCCTGCGGTCTGTGATTGGTTCTCGAGTGCAATGCTCCGCCCTGGGGCGGGG
</code></pre>
<p>We obtained the above via:</p>
<pre><code class="hljs css language-bash">efetch -db<span class="token operator">=</span>nuccore -id<span class="token operator">=</span>NC_000001.11 -format<span class="token operator">=</span>fasta -seq_start<span class="token operator">=</span><span class="token number">36306860</span> -seq_stop<span class="token operator">=</span><span class="token number">36307069</span>
</code></pre>
<p>This tool will be covered in later sections.</p>
<h3><a class="anchor" aria-hidden="true" id="enhancers"></a><a href="#enhancers" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Enhancers</h3>
<p>Enhancers help a gene turn on and regulate where, when, and how much the gene should be expressed for the cell to function properly. <strong>One enhancer can regulate many genes, and one gene can have many enhancers</strong>. Since DNA is folded and crumpled up in a cell, an enhancer can be far away from a gene it regulates in terms of distance measured along the DNA strand, but physically very close to the gene promoter. The best way to identify a sequence as an enhancer is to experimentally disrupt it and observe gene expression changes. Since this is cumbersome in practice, we use proxies like histone modifications (chemical decorations on the structural backbone upon which DNA is coiled) to identify enhancers. Silencers work similarly – they can be far from a gene, <strong>many-to-many relationships</strong>, hard to find – but cause a gene to be less, rather than more, expressed.</p>
<h2><a class="anchor" aria-hidden="true" id="what-is-homology"></a><a href="#what-is-homology" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>What is homology?</h2>
<p>Two regions of DNA that evolved from the same sequence (through processes of duplication of genomic regions and separation of two species) are <code>homologous</code>, or <code>homologs</code> of one another.</p>
<p>More specifically, regions in the genomes of two species that are descended from the same area in a common ancestor’s genome are <code>orthologs</code>. These regions are said to be <code>orthologous</code>.</p>
<p>On the other hand, <code>paralogous</code> sequences or <code>paralogs</code> were separated by duplication of a genomic region within the same genome.</p>
<p>Standard pet peeve of many scientists:</p>
<blockquote>
<p>Homology is not a synonym of sequence similarity!</p>
</blockquote>
<blockquote>
<p>Homologous sequences are usually similar to one another, but similarity of sequences does not indicate homology.</p>
</blockquote>
<p><img align="center" src="/home/docs/assets/img/BNF/homo-para-ortho.jpg"></img></p>
</span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/home/docs/BNF/getting_started/what-is-bioinformatics"><span class="arrow-prev">← </span><span>What is Bioinformatics?</span></a><a class="docs-next button" href="/home/docs/BNF/getting_started/how-is-bioinformatics-practiced"><span>How is Bioinformatics practiced?</span><span class="arrow-next"> →</span></a></div></div></div><nav class="onPageNav"><ul class="toc-headings"><li><a href="#biology-for-bioinformaticians">Biology for bioinformaticians</a></li><li><a href="#what-is-dna">What is DNA?</a><ul class="toc-headings"><li><a href="#what-is-dna-composed-of">What is DNA composed of?</a></li></ul></li><li><a href="#what-are-nucleotides">What are nucleotides?</a></li><li><a href="#what-are-base-pairs">What are base pairs?</a></li><li><a href="#what-are-dna-strands">What are DNA strands?</a></li><li><a href="#is-there-a-directionality-of-dna">Is there a directionality of DNA?</a></li><li><a href="#what-is-senseantisense">What is sense/antisense?</a></li><li><a href="#what-is-dna-sequencing">What is DNA sequencing?</a></li><li><a href="#what-gets-sequenced">What gets sequenced?</a></li><li><a href="#what-is-a-genome">What is a genome?</a></li><li><a href="#what-is-a-genomes-purpose">What is a genome’s purpose?</a></li><li><a href="#how-big-is-a-genome">How big is a genome?</a></li><li><a href="#what-is-rna">What is RNA?</a></li><li><a href="#how-does-a-genome-function">How does a genome function?</a></li><li><a href="#what-is-a-protein">What is a protein?</a></li><li><a href="#how-are-proteins-made">How are proteins made?</a></li><li><a href="#what-is-an-orf">What is an ORF?</a></li><li><a href="#what-is-a-gene">What is a gene?</a></li><li><a href="#do-genomes-have-other-features">Do genomes have other features?</a><ul class="toc-headings"><li><a href="#untranslated-regions">Untranslated regions</a></li><li><a href="#promoter-regions">Promoter regions</a></li><li><a href="#cpg-islands">CpG islands</a></li><li><a href="#enhancers">Enhancers</a></li></ul></li><li><a href="#what-is-homology">What is homology?</a></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="copyright">Copyright © 2021 Rajesh Detroja</section></footer></div><script>window.fbAsyncInit = function() {FB.init({appId:'351771359141773',xfbml:true,version:'v2.7'});};(function(d, s, id){var js, fjs = d.getElementsByTagName(s)[0];if (d.getElementById(id)) {return;}js = d.createElement(s); js.id = id;js.src = '//connect.facebook.net/en_US/sdk.js';fjs.parentNode.insertBefore(js, fjs);}(document, 'script','facebook-jssdk'));
                </script></body></html>